



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="PycoQC computes metrics and generates interactive QC plots for Oxford Nanopore technologies sequencing data">
      
      
        <link rel="canonical" href="https://a-slide.github.io/pycoQC/pycoQC/CLI_usage/">
      
      
        <meta name="author" content="Adrien Leger & Tommaso Leonardi">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.1, mkdocs-material-4.6.3">
    
    
      
        <title>Command line usage - pycoQC</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application.adb8469c.css">
      
      
    
    
      <script src="../../assets/javascripts/modernizr.86422ebf.js"></script>
    
    
      
    
    <link rel="stylesheet" href="../../assets/fonts/material-icons.css">
    
    
      <link rel="stylesheet" href="../../css/ansi-colours.css">
    
      <link rel="stylesheet" href="../../css/jupyter-cells.css">
    
      <link rel="stylesheet" href="../../css/pandas-dataframe.css">
    
    
      
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#pycoqc-cli-usage" tabindex="0" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://a-slide.github.io/pycoQC/" title="pycoQC" aria-label="pycoQC" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              pycoQC
            </span>
            <span class="md-header-nav__topic">
              
                Command line usage
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/a-slide/pycoQC/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    a-slide/pycoQC
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main" role="main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://a-slide.github.io/pycoQC/" title="pycoQC" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    pycoQC
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/a-slide/pycoQC/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    a-slide/pycoQC
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../installation/" title="Installation instructions" class="md-nav__link">
      Installation instructions
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3" checked>
    
    <label class="md-nav__link" for="nav-3">
      pycoQC Usage
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        pycoQC Usage
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../usage/" title="Using pycoQC" class="md-nav__link">
      Using pycoQC
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../API_usage/" title="jupyter API usage" class="md-nav__link">
      jupyter API usage
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Command line usage
      </label>
    
    <a href="./" title="Command line usage" class="md-nav__link md-nav__link--active">
      Command line usage
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#cli-usage" class="md-nav__link">
    CLI Usage
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#activate-virtual-environment" class="md-nav__link">
    Activate virtual environment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#getting-help" class="md-nav__link">
    Getting help
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#usage-examples" class="md-nav__link">
    Usage examples
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#basic-usage-quiet-mode" class="md-nav__link">
    Basic usage  (quiet mode)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#json-data-output-on-top-of-the-html-report" class="md-nav__link">
    JSON data output on top of the html report
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#including-guppy-barcoding-information" class="md-nav__link">
    Including guppy barcoding information
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#matching-multiple-files-with-a-regex-and-add-a-title-to-report" class="md-nav__link">
    Matching multiple files with a regex and add a title to report
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tweak-filtering-parameters" class="md-nav__link">
    Tweak filtering parameters
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#including-alignments-information-for-a-bam-file" class="md-nav__link">
    Including Alignments information for a Bam file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#advanced-configuration-with-custon-json-file" class="md-nav__link">
    Advanced configuration with custon json file
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      Barcode_split Usage
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Barcode_split Usage
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../Barcode_split/usage/" title="Using Barcode_split" class="md-nav__link">
      Using Barcode_split
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../Barcode_split/API_usage/" title="jupyter API usage" class="md-nav__link">
      jupyter API usage
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../Barcode_split/CLI_usage/" title="Command line usage" class="md-nav__link">
      Command line usage
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      Fast5_to_seq_summary Usage
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Fast5_to_seq_summary Usage
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../Fast5_to_seq_summary/usage/" title="Using Fast5_to_seq_summary" class="md-nav__link">
      Using Fast5_to_seq_summary
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../Fast5_to_seq_summary/API_usage/" title="jupyter API usage" class="md-nav__link">
      jupyter API usage
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../Fast5_to_seq_summary/CLI_usage/" title="Command line usage" class="md-nav__link">
      Command line usage
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../citing/" title="Citing pycoQC" class="md-nav__link">
      Citing pycoQC
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../acknowledgements/" title="Acknowledgements" class="md-nav__link">
      Acknowledgements
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../alternative/" title="Alternative and complementary packages" class="md-nav__link">
      Alternative and complementary packages
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../contributing/" title="Contributing Guidelines" class="md-nav__link">
      Contributing Guidelines
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../code_of_conduct/" title="Code of conduct" class="md-nav__link">
      Code of conduct
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../licence/" title="Licence" class="md-nav__link">
      Licence
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../changelog/" title="Changelog" class="md-nav__link">
      Changelog
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#cli-usage" class="md-nav__link">
    CLI Usage
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#activate-virtual-environment" class="md-nav__link">
    Activate virtual environment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#getting-help" class="md-nav__link">
    Getting help
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#usage-examples" class="md-nav__link">
    Usage examples
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#basic-usage-quiet-mode" class="md-nav__link">
    Basic usage  (quiet mode)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#json-data-output-on-top-of-the-html-report" class="md-nav__link">
    JSON data output on top of the html report
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#including-guppy-barcoding-information" class="md-nav__link">
    Including guppy barcoding information
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#matching-multiple-files-with-a-regex-and-add-a-title-to-report" class="md-nav__link">
    Matching multiple files with a regex and add a title to report
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tweak-filtering-parameters" class="md-nav__link">
    Tweak filtering parameters
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#including-alignments-information-for-a-bam-file" class="md-nav__link">
    Including Alignments information for a Bam file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#advanced-configuration-with-custon-json-file" class="md-nav__link">
    Advanced configuration with custon json file
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/a-slide/pycoQC/edit/master/docs/pycoQC/CLI_usage.ipynb" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h1 id="pycoqc-cli-usage">pycoQC CLI Usage</h1>
<p></div>
</div></p>
</div>
<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1">
PycoQC CLI can generate a beautiful HTML formatted report containing interactive D3.js plots. On top of it, the CLI can also dump summary information in a JSON formated file allowing easy parsing with third party tools.</p>
<p>The report is dynamically generated depending on the information available in the summary file.
</div>
</div></p>
</div>
<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h2 id="cli-usage">CLI Usage</h2>
<p></div>
</div></p>
</div>
<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h3 id="activate-virtual-environment">Activate virtual environment</h3>
<p></div>
</div></p>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="codehilite"><pre><span></span><code><span class="c1"># Using virtualenvwrapper here but can also be done with Conda </span>
workon pycoQC
</code></pre></div>

</div>

<div class="output_wrapper" markdown="1">
<div class="output" markdown="1">


<div class="output_area" markdown="1">
<div class="output_subarea output_stream output_stdout output_text">
<pre>(pycoQC) (pycoQC) </pre>
</div>
</div>

<div class="output_area" markdown="1">
<div class="output_subarea output_text output_error">
<pre></pre>
</div>
</div>

</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h3 id="getting-help">Getting help</h3>
<p></div>
</div></p>
</div>
<div class="cell border-box-sizing code_cell rendered"></div>
</div>
<div class="input">

<div class="codehilite"><pre><span></span><code>pycoQC -h
</code></pre></div>

</div>

<div class="output_wrapper" markdown="1">
<div class="output" markdown="1">


<div class="output_area" markdown="1">
<div class="output_subarea output_stream output_stdout output_text">
<pre>usage: pycoQC [-h] [--version]
              [--summary_file [SUMMARY_FILE [SUMMARY_FILE ...]]]
              [--barcode_file [BARCODE_FILE [BARCODE_FILE ...]]]
              [--bam_file [BAM_FILE [BAM_FILE ...]]]
              [--html_outfile HTML_OUTFILE] [--json_outfile JSON_OUTFILE]
              [--min_pass_qual MIN_PASS_QUAL] [--min_pass_len MIN_PASS_LEN]
              [--filter_calibration] [--filter_duplicated]
              [--min_barcode_percent MIN_BARCODE_PERCENT]
              [--report_title REPORT_TITLE] [--template_file TEMPLATE_FILE]
              [--config_file CONFIG_FILE] [--sample SAMPLE] [--default_config]
              [-v | -q]

pycoQC computes metrics and generates interactive QC plots from the sequencing summary
report generated by Oxford Nanopore technologies basecallers

* Minimal usage
    pycoQC -f sequencing_summary.txt -o pycoQC_output.html
* Including Guppy barcoding file + html output + json output
    pycoQC -f sequencing_summary.txt -b barcoding_sequencing.txt -o pycoQC_output.html -j pycoQC_output.json
* Including Bam file + html output
    pycoQC -f sequencing_summary.txt -a alignment.bam -o pycoQC_output.html

optional arguments:
  -h, --help            show this help message and exit
  --version             show program&#39;s version number and exit
  -v, --verbose         Increase verbosity
  -q, --quiet           Reduce verbosity

Input/output options:
  --summary_file [SUMMARY_FILE [SUMMARY_FILE ...]], -f [SUMMARY_FILE [SUMMARY_FILE ...]]
                        Path to a sequencing_summary generated by Albacore
                        1.0.0 + (read_fast5_basecaller.py) / Guppy 2.1.3+
                        (guppy_basecaller). One can also pass multiple space
                        separated file paths or a UNIX style regex matching
                        multiple files (Required)
  --barcode_file [BARCODE_FILE [BARCODE_FILE ...]], -b [BARCODE_FILE [BARCODE_FILE ...]]
                        Path to the barcode_file generated by Guppy 2.1.3+
                        (guppy_barcoder) or Deepbinner 0.2.0+. This is not a
                        required file. One can also pass multiple space
                        separated file paths or a UNIX style regex matching
                        multiple files (optional)
  --bam_file [BAM_FILE [BAM_FILE ...]], -a [BAM_FILE [BAM_FILE ...]]
                        Path to a Bam file corresponding to reads in the
                        summary_file. Preferably aligned with Minimap2 One can
                        also pass multiple space separated file paths or a
                        UNIX style regex matching multiple files (optional)
  --html_outfile HTML_OUTFILE, -o HTML_OUTFILE
                        Path to an output html file report (required if
                        json_outfile not given)
  --json_outfile JSON_OUTFILE, -j JSON_OUTFILE
                        Path to an output json file report (required if
                        html_outfile not given)

Filtering options:
  --min_pass_qual MIN_PASS_QUAL
                        Minimum quality to consider a read as &#39;pass&#39; (default:
                        7)
  --min_pass_len MIN_PASS_LEN
                        Minimum read length to consider a read as &#39;pass&#39;
                        (default: 0)
  --filter_calibration  If given, reads flagged as calibration strand by the
                        basecaller are removed (default: False)
  --filter_duplicated   If given, duplicated read_ids are removed but the
                        first occurence is kept (Guppy sometimes outputs the
                        same read multiple times) (default: False)
  --min_barcode_percent MIN_BARCODE_PERCENT
                        Minimal percent of total reads to retain barcode
                        label. If below, the barcode value is set as
                        `unclassified` (default: 0.1)

HTML report options:
  --report_title REPORT_TITLE
                        Title to use in the html report (default: PycoQC
                        report)
  --template_file TEMPLATE_FILE
                        Jinja2 html template for the html report (default: )
  --config_file CONFIG_FILE
                        Path to a JSON configuration file for the html report.
                        If not provided, looks for it in ~/.pycoQC and
                        ~/.config/pycoQC/config. If it&#39;s still not found,
                        falls back to default parameters. The first level keys
                        are the names of the plots to be included. The second
                        level keys are the parameters to pass to each plotting
                        function (default: )&#34;)

Other options:
  --sample SAMPLE       If not None a n number of reads will be randomly
                        selected instead of the entire dataset for ploting
                        function (deterministic sampling) (default: 100000)
  --default_config, -d  Print default configuration file. Can be used to
                        generate a template JSON file (default: False)
(pycoQC) </pre>
</div>
</div>

<div class="output_area" markdown="1">
<div class="output_subarea output_text output_error">
<pre></pre>
</div>
</div>

</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h3 id="usage-examples">Usage examples</h3>
<p></div>
</div></p>
</div>
<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h4 id="basic-usage-quiet-mode">Basic usage  (quiet mode)</h4>
<p></div>
</div></p>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="codehilite"><pre><span></span><code>pycoQC <span class="se">\</span>
    -f ./data/Albacore-1.2.1_basecall-1D-DNA_sequencing_summary.txt.gz <span class="se">\</span>
    -o ./results/Albacore-1.2.1_basecall-1D-DNA.html <span class="se">\</span>
    --quiet
</code></pre></div>

</div>

<div class="output_wrapper" markdown="1">
<div class="output" markdown="1">


<div class="output_area" markdown="1">
<div class="output_subarea output_stream output_stdout output_text">
<pre>Checking arguments values
Check input data files
Parse data files
Merge data
Cleaning data
Loading plotting interface
(pycoQC) </pre>
</div>
</div>

<div class="output_area" markdown="1">
<div class="output_subarea output_text output_error">
<pre></pre>
</div>
</div>

</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1">
<a href="https://a-slide.github.io/pycoQC/pycoQC/results/Albacore-1.2.1_basecall-1D-DNA.html">HTML OUTPUT</a>
</div>
</div></p>
</div>
<div class="cell border-box-sizing text_cell rendered"></div>
</div>
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h4 id="json-data-output-on-top-of-the-html-report">JSON data output on top of the html report</h4>
<p>A json report can be generated on top (or instead) of the html report</p>
<p>It contains a summarized version of the data collected by pycoQC in a structured and easy to parse format
</div>
</div></p>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="codehilite"><pre><span></span><code>pycoQC <span class="se">\</span>
    -f ./data/Guppy-2.1.3_basecall-1D-RNA_sequencing_summary.txt.gz <span class="se">\</span>
    -o ./results/Guppy-2.1.3_basecall-1D_RNA.html <span class="se">\</span>
    -j ./results/Guppy-2.1.3_basecall-1D_RNA.json <span class="se">\</span>
    --quiet
</code></pre></div>

</div>

<div class="output_wrapper" markdown="1">
<div class="output" markdown="1">


<div class="output_area" markdown="1">
<div class="output_subarea output_stream output_stdout output_text">
<pre>Checking arguments values
Check input data files
Parse data files
Merge data
Cleaning data
Loading plotting interface
(pycoQC) </pre>
</div>
</div>

<div class="output_area" markdown="1">
<div class="output_subarea output_text output_error">
<pre></pre>
</div>
</div>

</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1">
<a href="https://a-slide.github.io/pycoQC/pycoQC/results/Guppy-2.1.3_basecall-1D_RNA.html">HTML OUTPUT</a></p>
<p><a href="https://a-slide.github.io/pycoQC/pycoQC/results/Guppy-2.1.3_basecall-1D_RNA.json">JSON OUTPUT</a>
</div>
</div></p>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h4 id="including-guppy-barcoding-information">Including guppy barcoding information</h4>
<p></div>
</div></p>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="codehilite"><pre><span></span><code>pycoQC <span class="se">\</span>
    -f ./data/Guppy-2.1.3_basecall-1D-DNA_sequencing_summary.txt.gz <span class="se">\</span>
    -b ./data/Guppy-2.1.3_basecall-1D_DNA_barcoding_summary.txt.gz <span class="se">\</span>
    -o ./results/Guppy-2.1.3_basecall-1D_DNA_barcode.html <span class="se">\</span>
    --quiet
</code></pre></div>

</div>

<div class="output_wrapper" markdown="1">
<div class="output" markdown="1">


<div class="output_area" markdown="1">
<div class="output_subarea output_stream output_stdout output_text">
<pre>Checking arguments values
Check input data files
Parse data files
Merge data
Cleaning data
Loading plotting interface
(pycoQC) </pre>
</div>
</div>

<div class="output_area" markdown="1">
<div class="output_subarea output_text output_error">
<pre></pre>
</div>
</div>

</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1">
<a href="https://a-slide.github.io/pycoQC/pycoQC/results/Guppy-2.1.3_basecall-1D_DNA_barcode.html">HTML OUTPUT</a>
</div>
</div></p>
</div>
<div class="cell border-box-sizing text_cell rendered"></div>
</div>
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h4 id="matching-multiple-files-with-a-regex-and-add-a-title-to-report">Matching multiple files with a regex and add a title to report</h4>
<p></div>
</div></p>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="codehilite"><pre><span></span><code>pycoQC <span class="se">\</span>
    -f ./data/Albacore*RNA* <span class="se">\</span>
    -o ./results/Albacore_all_RNA.html <span class="se">\</span>
    --report_title <span class="s2">&quot;All RNA runs&quot;</span> <span class="se">\</span>
    --quiet
</code></pre></div>

</div>

<div class="output_wrapper" markdown="1">
<div class="output" markdown="1">


<div class="output_area" markdown="1">
<div class="output_subarea output_stream output_stdout output_text">
<pre>Checking arguments values
Check input data files
Parse data files
Merge data
Cleaning data
Loading plotting interface
(pycoQC) </pre>
</div>
</div>

<div class="output_area" markdown="1">
<div class="output_subarea output_text output_error">
<pre></pre>
</div>
</div>

</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1">
<a href="https://a-slide.github.io/pycoQC/pycoQC/results/Albacore_all_RNA.html">HTML OUTPUT</a>
</div>
</div></p>
</div>
<div class="cell border-box-sizing text_cell rendered"></div>
</div>
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h4 id="tweak-filtering-parameters">Tweak filtering parameters</h4>
<p></div>
</div></p>
<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1">
* Define reads with a quality higher than 8 and length higher than 200 bases as "pass" 
* Discard reads aligned on the calibration standard
* Unset value of any barcode found in less than 10% of the reads
</div>
</div></p>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="codehilite"><pre><span></span><code>pycoQC <span class="se">\</span>
    -f ./data/Albacore-2.1.10_basecall-1D-DNA_sequencing_summary.txt.gz<span class="se">\</span>
    -o ./results/Albacore-2.1.10_basecall-1D-DNA.html <span class="se">\</span>
    --min_pass_qual <span class="m">8</span> <span class="se">\</span>
    --min_pass_len <span class="m">200</span> <span class="se">\</span>
    --filter_calibration <span class="se">\</span>
    --min_barcode_percent <span class="m">10</span> <span class="se">\</span>
    --quiet
</code></pre></div>

</div>

<div class="output_wrapper" markdown="1">
<div class="output" markdown="1">


<div class="output_area" markdown="1">
<div class="output_subarea output_stream output_stdout output_text">
<pre>Checking arguments values
Check input data files
Parse data files
Merge data
Cleaning data
Loading plotting interface
(pycoQC) </pre>
</div>
</div>

<div class="output_area" markdown="1">
<div class="output_subarea output_text output_error">
<pre></pre>
</div>
</div>

</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1">
<a href="https://a-slide.github.io/pycoQC/pycoQC/results/Albacore-2.1.10_basecall-1D-DNA.html">HTML OUTPUT</a>
</div>
</div></p>
</div>
<div class="cell border-box-sizing text_cell rendered"></div>
</div>
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h4 id="including-alignments-information-for-a-bam-file">Including Alignments information for a Bam file</h4>
<p></div>
</div></p>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="codehilite"><pre><span></span><code>pycoQC <span class="se">\</span>
    -f ./large_data/sample_1_sequencing_summary.txt <span class="se">\</span>
    -a ./large_data/sample_1.bam <span class="se">\</span>
    -o ./results/Guppy-2.3_basecall-1D_alignment-DNA.html <span class="se">\</span>
    -j ./results/Guppy-2.3_basecall-1D_alignment-DNA.json <span class="se">\</span>
    --quiet
</code></pre></div>

</div>

<div class="output_wrapper" markdown="1">
<div class="output" markdown="1">


<div class="output_area" markdown="1">
<div class="output_subarea output_stream output_stdout output_text">
<pre>Checking arguments values
Check input data files
Parse data files
Merge data
Cleaning data
Loading plotting interface
(pycoQC) </pre>
</div>
</div>

<div class="output_area" markdown="1">
<div class="output_subarea output_text output_error">
<pre></pre>
</div>
</div>

</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1">
<a href="https://a-slide.github.io/pycoQC/pycoQC/results/Guppy-2.3_basecall-1D_alignment-DNA.html">HTML OUTPUT</a></p>
<p><a href="https://a-slide.github.io/pycoQC/pycoQC/results/Guppy-2.3_basecall-1D_alignment-DNA.json">JSON OUTPUT</a>
</div>
</div></p>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1"></p>
<h4 id="advanced-configuration-with-custon-json-file">Advanced configuration with custon json file</h4>
<p></div>
</div></p>
</div>
<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1">
Although we recommend to stick to the default parameters, a json formatted configuration file can be provided to tweak the plots. A default configuration file can be generated using:
</div>
</div></p>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="codehilite"><pre><span></span><code>pycoQC --default_config
</code></pre></div>

</div>

<div class="output_wrapper" markdown="1">
<div class="output" markdown="1">


<div class="output_area" markdown="1">
<div class="output_subarea output_stream output_stdout output_text">
<pre>{
  &#34;run_summary&#34;: {
    &#34;plot_title&#34;: &#34;General run summary&#34;
  },
  &#34;basecall_summary&#34;: {
    &#34;plot_title&#34;: &#34;Basecall summary&#34;
  },
  &#34;alignment_summary&#34;: {
    &#34;plot_title&#34;: &#34;Alignment summary&#34;
  },
  &#34;read_len_1D&#34;: {
    &#34;plot_title&#34;: &#34;Basecalled reads length&#34;,
    &#34;color&#34;: &#34;lightsteelblue&#34;,
    &#34;nbins&#34;: 200,
    &#34;smooth_sigma&#34;: 2
  },
  &#34;align_len_1D&#34;: {
    &#34;plot_title&#34;: &#34;Aligned reads length&#34;,
    &#34;color&#34;: &#34;mediumseagreen&#34;,
    &#34;nbins&#34;: 200,
    &#34;smooth_sigma&#34;: 2
  },
  &#34;read_qual_1D&#34;: {
    &#34;plot_title&#34;: &#34;Basecalled reads PHRED quality&#34;,
    &#34;color&#34;: &#34;salmon&#34;,
    &#34;nbins&#34;: 200,
    &#34;smooth_sigma&#34;: 2
  },
  &#34;identity_freq_1D&#34;: {
    &#34;plot_title&#34;: &#34;Aligned reads identity&#34;,
    &#34;color&#34;: &#34;sandybrown&#34;,
    &#34;nbins&#34;: 200,
    &#34;smooth_sigma&#34;: 2
  },
  &#34;read_len_read_qual_2D&#34;: {
    &#34;plot_title&#34;: &#34;Basecalled reads length vs reads PHRED quality&#34;,
    &#34;x_nbins&#34;: 200,
    &#34;y_nbins&#34;: 100,
    &#34;smooth_sigma&#34;: 2
  },
  &#34;read_len_align_len_2D&#34;: {
    &#34;plot_title&#34;: &#34;Basecalled reads length vs alignments length&#34;,
    &#34;x_nbins&#34;: 200,
    &#34;y_nbins&#34;: 100,
    &#34;smooth_sigma&#34;: 1
  },
  &#34;align_len_identity_freq_2D&#34;: {
    &#34;plot_title&#34;: &#34;Aligned reads length vs alignments identity&#34;,
    &#34;x_nbins&#34;: 200,
    &#34;y_nbins&#34;: 100,
    &#34;smooth_sigma&#34;: 2
  },
  &#34;read_qual_identity_freq_2D&#34;: {
    &#34;plot_title&#34;: &#34;Reads PHRED quality vs alignments identity&#34;,
    &#34;x_nbins&#34;: 200,
    &#34;y_nbins&#34;: 100,
    &#34;smooth_sigma&#34;: 1
  },
  &#34;output_over_time&#34;: {
    &#34;plot_title&#34;: &#34;Output over experiment time&#34;,
    &#34;cumulative_color&#34;: &#34;rgb(204,226,255)&#34;,
    &#34;interval_color&#34;: &#34;rgb(102,168,255)&#34;
  },
  &#34;read_len_over_time&#34;: {
    &#34;plot_title&#34;: &#34;Read length over experiment time&#34;,
    &#34;median_color&#34;: &#34;rgb(102,168,255)&#34;,
    &#34;quartile_color&#34;: &#34;rgb(153,197,255)&#34;,
    &#34;extreme_color&#34;: &#34;rgba(153,197,255,0.5)&#34;,
    &#34;smooth_sigma&#34;: 1
  },
  &#34;read_qual_over_time&#34;: {
    &#34;plot_title&#34;: &#34;Read quality over experiment time&#34;,
    &#34;median_color&#34;: &#34;rgb(250,128,114)&#34;,
    &#34;quartile_color&#34;: &#34;rgb(250,170,160)&#34;,
    &#34;extreme_color&#34;: &#34;rgba(250,170,160,0.5)&#34;,
    &#34;smooth_sigma&#34;: 1
  },
  &#34;align_len_over_time&#34;: {
    &#34;plot_title&#34;: &#34;Aligned reads length over experiment time&#34;,
    &#34;median_color&#34;: &#34;rgb(102,168,255)&#34;,
    &#34;quartile_color&#34;: &#34;rgb(153,197,255)&#34;,
    &#34;extreme_color&#34;: &#34;rgba(153,197,255,0.5)&#34;,
    &#34;smooth_sigma&#34;: 1
  },
  &#34;identity_freq_over_time&#34;: {
    &#34;plot_title&#34;: &#34;Aligned reads identity over experiment time&#34;,
    &#34;median_color&#34;: &#34;rgb(250,128,114)&#34;,
    &#34;quartile_color&#34;: &#34;rgb(250,170,160)&#34;,
    &#34;extreme_color&#34;: &#34;rgba(250,170,160,0.5)&#34;,
    &#34;smooth_sigma&#34;: 1
  },
 &#34;barcode_counts&#34;: {
    &#34;plot_title&#34;: &#34;Number of reads per barcode&#34;,
    &#34;colors&#34;: [
      &#34;#f8bc9c&#34;,
      &#34;#f6e9a1&#34;,
      &#34;#f5f8f2&#34;,
      &#34;#92d9f5&#34;,
      &#34;#4f97ba&#34;
    ]
  },
  &#34;channels_activity&#34;: {
    &#34;plot_title&#34;: &#34;Channel activity over time&#34;,
    &#34;smooth_sigma&#34;: 1
  },
  &#34;alignment_reads_status&#34;: {
    &#34;plot_title&#34;: &#34;Summary of reads alignment&#34;,
    &#34;colors&#34;: [
      &#34;#f8bc9c&#34;,
      &#34;#f6e9a1&#34;,
      &#34;#92d9f5&#34;,
      &#34;#4f97ba&#34;,
      &#34;#f5f8f2&#34;
    ]
  },
  &#34;alignment_rate&#34;: {
    &#34;plot_title&#34;: &#34;Bases alignment rate&#34;,
    &#34;colors&#34;: [
      &#34;#fcaf94&#34;,
      &#34;#828282&#34;,
      &#34;#fc8161&#34;,
      &#34;#828282&#34;,
      &#34;#f44f39&#34;,
      &#34;#d52221&#34;,
      &#34;#828282&#34;,
      &#34;#828282&#34;,
      &#34;#828282&#34;,
      &#34;#828282&#34;
    ]
  },
  &#34;alignment_coverage&#34;: {
    &#34;plot_title&#34;: &#34;Coverage overview&#34;,
    &#34;nbins&#34;: 500,
    &#34;color&#34;: &#34;rgba(102,168,255,0.75)&#34;,
    &#34;smooth_sigma&#34;: 1
  }
}
(pycoQC) </pre>
</div>
</div>

<div class="output_area" markdown="1">
<div class="output_subarea output_text output_error">
<pre></pre>
</div>
</div>

</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1">
To save and edit it redirect the std output to a file and make your changes using your favorite text editor.</p>
<p>To remove a plot from the report, just remove it (or comment it) from the configuration file</p>
</div>
</div>
<p>The configuration file accept all the arguments of the target plotting functions. For more information refer to the API documentation
</div>
</div></p>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="codehilite"><pre><span></span><code>pycoQC --default_config &gt; data/pycoQC_config.json
</code></pre></div>

</div>

<div class="output_wrapper" markdown="1">
<div class="output" markdown="1">


<div class="output_area" markdown="1">
<div class="output_subarea output_stream output_stdout output_text">
<pre>(pycoQC) </pre>
</div>
</div>

<div class="output_area" markdown="1">
<div class="output_subarea output_text output_error">
<pre></pre>
</div>
</div>

</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1">
Run pycoQC with <code class="codehilite"><span class="err">--config</span></code> option
</div>
</div></p>
</div>
<div class="cell border-box-sizing code_cell rendered"></div>
</div>
<div class="input">

<div class="codehilite"><pre><span></span><code>pycoQC <span class="se">\</span>
    -f ./data/Albacore-1.7.0_basecall-1D-DNA_sequencing_summary.txt.gz<span class="se">\</span>
    -o ./results/Albacore-1.7.0_basecall-1D-DNA.html <span class="se">\</span>
    --config ./data/pycoQC_config.json <span class="se">\</span>
    --quiet
</code></pre></div>

</div>

<div class="output_wrapper" markdown="1">
<div class="output" markdown="1">


<div class="output_area" markdown="1">
<div class="output_subarea output_stream output_stdout output_text">
<pre>Checking arguments values
Check input data files
Parse data files
Merge data
Cleaning data
Loading plotting interface
(pycoQC) </pre>
</div>
</div>

<div class="output_area" markdown="1">
<div class="output_subarea output_text output_error">
<pre></pre>
</div>
</div>

</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<p><div class="inner_cell" markdown="1">
<div class="text_cell_render border-box-sizing rendered_html" markdown="1">
<a href="https://a-slide.github.io/pycoQC/pycoQC/results/Albacore-1.7.0_basecall-1D-DNA.html">HTML OUTPUT</a>
</div>
</div></p>
</div>
                
                  
                
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../API_usage/" title="jupyter API usage" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                jupyter API usage
              </span>
            </div>
          </a>
        
        
          <a href="../../Barcode_split/usage/" title="Using Barcode_split" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Using Barcode_split
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2019 Adrien Leger & Tommaso Leonardi 
          </div>
        
        powered by
        <a href="https://www.mkdocs.org" target="_blank" rel="noopener">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs</a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application.c33a9706.js"></script>
      
      <script>app.initialize({version:"1.1",url:{base:"../.."}})</script>
      
    
  </body>
</html>